starlite/utils/predicates.py

```diff
@@ -66,4 +77,40 @@ def is_optional_union(annotation: Any) -> bool:
     Returns:
         True for a union, False otherwise.
     """
-    return get_origin(annotation) in UNION_TYPES and type(None) in get_args(annotation)
+    return get_origin(annotation) in UNION_TYPES and NoneType in get_args(annotation)
+
+
+def is_dataclass_type_typeguard(value: Any) -> "TypeGuard[DataclassType]":
+    """Wrapper for `is_dataclass()` that narrows to type only, not instance.
+
+    Args:
+        value: tested to determine if type of `dataclass`.
+
+    Returns:
+        `True` if `value` is a `dataclass` type.
+    """
+    return is_dataclass(value) and isinstance(value, type)
+
+
+def is_dataclass_type_or_instance_typeguard(value: Any) -> "TypeGuard[DataclassTypeOrInstance]":
```

Comment:
We could instead have `is_dataclass_instance_typeguard` and `is_dataclass_class_typeguard`

Reply:
yep went with "class" instead of "type"